src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrBinaryFileScanner.cs

```diff
@@ -72,12 +73,31 @@ private async Task<IEnumerable<string>> ScanAndParseResult(string[] args, string
 
         ret.ThrowIfFailed($"Failed to scan the {repoName} repository");
 
-        // We want to exlude empty files from the list. These empty files still have a few chracters that we want to ignore
+        // Git marks UTF 16 text files as binary, we want to exclude these
         return ret.StandardOutput
             .Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries)
             .Where(line => line.StartsWith(BinaryFileMarker))
             .Select(line => line.Split('\t').Last())
-            .Where(file => new FileInfo(_vmrInfo.VmrPath / file).Length > 6);
+            .ToAsyncEnumerable()
+            .WhereAwait(async file => await IsNotUTF16(_vmrInfo.VmrPath / file))
+            .ToEnumerable();
+    }
+
+    private async Task<bool> IsNotUTF16(LocalPath filePath)
+    {
+        if (Environment.OSVersion.Platform == PlatformID.Unix)
+        {
+            var ret = await _processManager.Execute(executable: "file", arguments: new string[] { filePath });
```

Comment:
I think this might work as well thanks to the overload:
```suggestion
            var ret = await _processManager.Execute("file", filePath);
```